src/fem/fe_values_views.hh

```diff
@@ -0,0 +1,156 @@
+/*!
+ *
+﻿ * Copyright (C) 2015 Technical University of Liberec.  All rights reserved.
+ * 
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 3 as published by the
+ * Free Software Foundation. (http://www.gnu.org/licenses/gpl-3.0.en.html)
+ * 
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+ *
+ * 
+ * @file    fe_system.hh
+ * @brief   Class FESystem for compound finite elements.
+ * @author  Jan Stebel
+ */
+
+#ifndef FE_VALUES_VIEWS_HH
+#define FE_VALUES_VIEWS_HH
+
+#include <vector>
+#include <armadillo>
+
+template<unsigned int dim, unsigned int spacedim> class FEValuesBase;
+
+
+/**
+ * FEValuesExtractors are auxiliary classes that hold information
+ * about the particular scalar component or offset of the first vector,
+ * tensor etc. component of the finite element.
+ * 
+ * The classes are motivated by deal.II.
+ */
+namespace FEValuesExtractors {
+
+  class Scalar {
+  public:
+    Scalar(unsigned int comp) : component_(comp) {};
+    
+    unsigned int component_;
+  };
+  
+  class Vector {
+  public:
+    Vector(unsigned int comp) : first_vector_component_(comp) {};
+    
+    unsigned int first_vector_component_;
+  };
+  
```

Comment:
add empty Tensor

Reply:
Make methods: FeValues<>::get_scalar_view, get_vector_view, get_tensor_view which takes component index as parameter and return particular extractor. This allows performing checking of compatibility of component with the system structure. Add specification of quantinty in constructor of FeSystem, optional parameter saying that I;m creating Vector or Tensor quantity.

Reply:
If I understand well, the new methods should have names get_{scalar,vector,tensor}_extractor (not view). Then I have 2 objections:
- the FEValues object must exist before the extractor
- the extractor does not change for different FEValues when using the same FiniteElement and hence it is more natural to place these new methods into FiniteElement